cli/command/stack/deploy.go

```diff
@@ -62,6 +71,18 @@ func runDeploy(dockerCli command.Cli, opts deployOptions) error {
 	}
 }
 
+// checkImageResolutionAllowed validates the opts.resolveImage command line option
+// and also turns image resolution off if the version is older than 1.30
+func checkImageResolutionAllowed(dockerCli command.Cli, opts *deployOptions) error {
+	if opts.resolveImage != "always" && opts.resolveImage != "changed" && opts.resolveImage != "never" {
```

Comment:
I'd suggest declaring constants for the possible values, so it is harder for this to be out of sync with the code that interprets the flag values, or for typos to creep into the string literals in some places. Again, not a big deal either way.